### PR TITLE
carli/2034_single_source_catalog_file_not_rendering

### LINKED ISSUE #2034: missing catalog image overlay when there is only one source in the catalog file
**Describe the bug**
It appears that if a catalog file has only one source, if we have two images matched and enable the catalog image overlay, we will only see one image has the catalog overlay. The other is empty. If we load a catalog with two sources instead and repeat the exercise, we will see two sources on both images as expected.

**To Reproduce**
1. download the test images and test catalog files
https://drive.google.com/file/d/164C3NjSyE30T2BHXi7Y9FaTFm7AzfDc7/view?usp=sharing

2. load the two images and match them
3. load catalog.vot which contains only one source and enable catalog overlay. We will see the issue.
4. repeat the test with catalog2.vot which contains two sources. We will see the expected behavior.

**Expected behavior**
with the single source catalog file, we should see the catalog overlay on all matched images.

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS Monterey
 - Browser [e.g. chrome, safari, electron app]: electron app
 - Browser version [e.g. 22]: 
 - Backend branch [e.g. dev, v3b2 release]: v3-stable
 - Frontend branch [e.g. dev, v3b2 release]: v3-stable

### PR DESCRIPTION
**Pull request by crocka**
**Description**

The problem is from `controlMapLookUp` in `utilities.glsl`. The `index` is calculated with `range`, but when there is only one source, `range` is 0, which results in the wrong `index`. The resolution is to add an if-statement to account for when `range` is 0. This PR fixes #2034.

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added~
- [x] ~~changelog updated~~ / no changelog update needed
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

File: src/services/GLSL/utilities.glsl [hunk 1]
```diff
@@ -75,6 +75,9 @@ vec2 controlMapLookup(sampler2D controlMapTexture, vec2 pos, vec2 controlMapSize
     vec2 range = controlMapMax - controlMapMin;
     vec2 shiftedPoint = pos - controlMapMin;
     vec2 index = shiftedPoint / range * controlMapSize;
+    if (controlMapMax == controlMapMin) {
```
Comment (crocka) [inline src/services/GLSL/utilities.glsl]: I tried
```if (range.x == 0 && range.y == 0) { index = vec2(0, 0) };```
but it didn't seem to work. I am not very familiar with the syntax of GLSL, so please advice if there is a better solution.

Comment (kswang1029) [reply src/services/GLSL/utilities.glsl]: The current solution works fine. No regression is observed from e2e tests. If there are better solutions from @YuHsuan-Hwang , I will re-do the test.

Comment (YuHsuan-Hwang) [reply src/services/GLSL/utilities.glsl]: @crocka Does `if (range.x == 0.0 && range.y == 0.0) { index = vec2(0, 0) };` works?
`controlMapMax == controlMapMin` is also fine.

Comment (crocka) [reply src/services/GLSL/utilities.glsl]: This is actually the first thing I tried. It didn't work and it is giving me the following error.
Screenshot from 2023-01-03 20-07-20

Comment (YuHsuan-Hwang) [reply src/services/GLSL/utilities.glsl]: Did you use `0.0` or `0`? You can check the console for error messages.

Comment (crocka) [reply src/services/GLSL/utilities.glsl]: I see the problem. We should use range.x == 0.0 && range.y == 0.0. I tried `index = vec2(0.0, 0.0) but ignored the if-statement.

**Review by YuHsuan-Hwang — APPROVED** [review]
need a space after `if`. Other than that this looks good.
I think this is a regression from my control map border fix PR #1870. Thanks for fixing again 👍  Please also update the change log.

@kswang1029 We are ready for a final feature test, thanks.

**Review by kswang1029 — APPROVED** [review]

**Pull request merged**